tests/integration/deploy_test.go

```diff
@@ -0,0 +1,418 @@
+				"--budget-notification-from-email", "from@example.com",
+				"--dce-version", "v0.12.3",
+			},
+			expectedMainTf: []string{
+				`
+variable "aws_region" {
+	type = string
+	default = "moon-darkside-1"
+}`,
+				`
+variable "namespace" {
+	type = string
+	default = "my-namespace"
+}`,
+				`
+variable "budget_notification_from_email" {
+	type = string
+	default = "from@example.com"
+}`,
+				`
+module "dce" {
+	source="github.com/Optum/dce//modules?ref=v0.12.3"
+
+	aws_region = var.aws_region
+	namespace = var.namespace
+	budget_notification_from_email = var.budget_notification_from_email
+}`,
+			},
+			expectedDeployedVersion: "0.12.3",
+		})
+	})
+
+	t.Run("should accept tfvars as YAML config", func(t *testing.T) {
+		deployCommandTestCase(t, &deployTestCase{
+			cliCommand: []string{"system", "deploy",},
+			yamlConfig: &configs.Root{
+				Deploy: configs.Deploy{
+					Version:                     stringp("v0.12.3"),
+					AWSRegion:                   stringp("moon-darkside-1"),
+					Namespace:                   stringp("my-namespace"),
+					BudgetNotificationFromEmail: stringp("from@example.com"),
+				},
+			},
+			expectedMainTf: []string{
+				`
+variable "aws_region" {
+	type = string
+	default = "moon-darkside-1"
+}`,
+				`
+variable "namespace" {
+	type = string
+	default = "my-namespace"
+}`,
+				`
+variable "budget_notification_from_email" {
+	type = string
+	default = "from@example.com"
+}`,
+				`
+module "dce" {
+	source="github.com/Optum/dce//modules?ref=v0.12.3"
+
+	aws_region = var.aws_region
+	namespace = var.namespace
+	budget_notification_from_email = var.budget_notification_from_email
+}`,
+			},
+			expectedDeployedVersion: "0.12.3",
+		})
+	})
+
+	t.Run("should accept env var configuration", func(t *testing.T) {
+		deployCommandTestCase(t, &deployTestCase{
+			cliCommand: []string{"system", "deploy",},
+			envVars: map[string]string{
+				"DCE_VERSION":                        "v9999.12.3",
+				"AWS_REGION":                         "moon-darkside-1",
+				"DCE_NAMESPACE":                      "my-namespace",
+				"DCE_BUDGET_NOTIFICATION_FROM_EMAIL": "from@example.com",
+			},
+			expectedMainTf: []string{
+				`
+variable "aws_region" {
+	type = string
+	default = "moon-darkside-1"
+}`,
+				`
+variable "namespace" {
+	type = string
+	default = "my-namespace"
+}`,
+				`
+variable "budget_notification_from_email" {
+	type = string
+	default = "from@example.com"
+}`,
+				`
+module "dce" {
+	source="github.com/Optum/dce//modules?ref=v9999.12.3"
+
+	aws_region = var.aws_region
+	namespace = var.namespace
+	budget_notification_from_email = var.budget_notification_from_email
+}`,
+			},
+			expectedDeployedVersion: "9999.12.3",
+		})
+	})
+
+	t.Run("should not prompt for approval, if batch mode is disabled", func(t *testing.T) {
+		deployCommandTestCase(t, &deployTestCase{
+			cliCommand:             []string{"system", "deploy", "--batch-mode"},
+			expectDeploymentPrompt: boolp(false),
+		})
+	})
+}
```

Comment:
There's a lot more that could be tested here, but I just wanted to get things rolling.